Seed vault - strange seed design disk & biogenerator imports

FantasticFwoosh opened the pull request:
:cl: Benevolent Moonman
add: The lifebringers have left behind a particularly special design disk for a unusual plant utilizing advanced technology to turn biomatter goop into life.
add: Biogenerators now have a import category for relevant additions from design disks.
/:cl:

The Biogenerator can now support imported designs through its existing design disk functionality, which had no previous use in order to use specially placed seed vault loot design disk that creates strange seeds at a cost of (**5000**) biogenerated content in order to extend gameplay and roll for good random chems (in which the odds are entirely on luck)

- The station biogenerator can also use this disk once it is taken from the seed vault, but it is still laborious to use as per top tier femto manipulator parts only decreases the cost to (**1250**) biogenerator content which is manageable but still expensive without optimised large yields of plant matter input.

What plant chems will you discover? 🌱

FantasticFwoosh committed:
Updates seed vault map & biogenerator additions

Biogenerator can now support imported designs through its existing
design disk functionality. A specially made disk added into the seed
vault allows vault dwellers to use this feature also.

Repukan commented:
FUCK NO

I DONT WANT FUNGAL TB IN LAVALAND

FUCK OFF FWOOSH

Cyberboss commented on _maps/RandomRuins/LavaRuins/lavaland_surface_seed_vault.dmm:
```diff
@@ -1,708 +1,69 @@
-/obj/effect/baseturf_helper/lava_land/surface,
-/turf/closed/wall/r_wall,
-/area/ruin/powered/seedvault)
+"a" = (/turf/closed/mineral/volcanic/lava_land_surface,/area/lavaland/surface/outdoors)
```
TGM

FantasticFwoosh replied:
Sorry please explain to me, map edits aren't usually my thing.

Cyberboss replied:
I'm saying this needs to be run through the map merge tool: https://tgstation13.org/wiki/Map_Merger

FantasticFwoosh commented:
Don't let diseased immigrants rush to the station by beelining then @Repukan the risk isn't exaggerated that much since it takes a while to grind through lots of strange seeds

ma44 commented:
Besides having a 60% chance to actually get a random gene, on top of that there are several hundred reagents, this sounds like a good lottery thing
👍

FantasticFwoosh committed:
Merge Fix

moved the disk a little bit without consequence to hopefullly rectify
with map merger tools.

GunHog commented:
Hardcore Lavaland miners do love their super high difficulty levels...

Cruix commented on code/modules/ruins/lavaland_ruin_code.dm:
```diff
@@ -16,6 +16,27 @@
+	build_type = BIOGENERATOR
+	materials = list(MAT_BIOMASS = 5000)
+	build_path = /obj/item/seeds/random
+	category = list("initial", "Imported")
```
If it has "initial" in the category, all biogenerators will start with this by default.

FantasticFwoosh replied:
It is empty until a design disk with something compatible also with "imported" is applied, and this is intentional futureproofing, else the swiped contents are going to be added to the rest of the categories.

- The same way the autolathe also has a constantly empty until filled "imported" section for ported over RnD and golem disk stuff.

Cruix replied:
https://github.com/tgstation/tgstation/blob/af4d9a85c9a3eeef1e5e0637ef4ebd0312794ac7/code/modules/research/research.dm#L170
If a design has "initial" in the category var and the BIOGENERATOR flag in build_type, it will appear at roundstart in all biogenerators. Notice how the golem shell design, right below this, does not have "initial" in its category var.

FantasticFwoosh replied:
I have tested it, it works regardlessly but if you want me to **remove** "initial" on the basis of it being redundant just say so please.

- I will test to make sure it isn't simultaneously spread to all over biogenerators once the disk is swiped into the biogenerator just to be sure, but i can already confirm that initial is empty on roundstart.

Cruix replied:
I have pulled your branch and tested it, and the strange seed does appear in the botany biogenerator at roundstart.

Cruix commented on _maps/RandomRuins/LavaRuins/lavaland_surface_seed_vault.dmm:
```diff
@@ -1,708 +1,69 @@
+"e" = (/obj/structure/table/wood,/obj/item/weapon/storage/box/disks_plantgene,/obj/machinery/light{dir = 1},/turf/open/floor/plasteel/freezer,/area/ruin/powered/seedvault)
+"f" = (/obj/structure/table/wood,/obj/item/weapon/storage/box/disks_plantgene,/turf/open/floor/plasteel/freezer,/area/ruin/powered/seedvault)
+"g" = (/obj/machinery/plantgenes/seedvault,/turf/open/floor/plasteel/freezer,/area/ruin/powered/seedvault)
+"h" = (/obj/structure/table/wood,/obj/item/weapon/disk/design_disk/strange_seed{step_x = 0; step_y = 1},/turf/open/floor/plasteel/freezer,/area/ruin/powered/seedvault)
```
Change step_x and step_y to pixel_x and pixel_y, otherwise gliding will be globally disabled whenever this ruin is loaded.

FantasticFwoosh replied:
i did that purely so i could (hopefully) get issues id been having with the map merger working again if you read the small print, im on my way to rectifying it.

FantasticFwoosh committed:
Fixes "initial" and #2 attempt at map conflict

Hopefully allows the map to work with the disk loot removed and put back
into original place to constitute a change, initial has been omitted
from the strange seed design disk.

FantasticFwoosh commented:
Im getting progressively annoyed at map merger even though the code is in place and positioned. I might put this on ice but ill give it till the rest of today to try and explore ways for me to get it to work.

kevinz000 commented:
>more botany buffs and incentive to kill lifebringers

FantasticFwoosh commented:
Biogenerator designs are a buff for everyone including the garden, i have the code stored but ill have to upgrade my understanding and how i use git before i do map-stuff. Stick to hard code for now since that's the only real thing that works out for me.
